public/docs/ts/latest/tutorial/toh-pt6.jade

```diff
@@ -0,0 +1,252 @@
+
+:marked
+  Here is `HeroDetailComponent` with the added save button.
+  
+figure.image-display
+  img(src='/resources/images/devguide/toh/hero-details-save-button.png' alt="Hero Details With Save Button") 
+
+:marked
+  ### Delete
+  
+  We have added the option to delete hereos from `HeroesComponent`. `HeroService` will delete the hero, but we still have to filter out the deleted hero from the list to update the view.
+  
++makeExample('toh-6/ts/app/heroes.component.ts', 'delete', 'app/heroes.component.ts (delete)')(format=".")  
+
+:marked
+  Here is `HeroesComponent` with the delete button. 
+  
+figure.image-display
+  img(src='/resources/images/devguide/toh/heroes-list-delete-button.png' alt="Heroes List With Delete Button")  
+
+.l-main-section  
+:marked
+ ## Integration with Http
+  
+  `Http` is not part of Angular core, but exists as a separate add-on module called `angular2/http`. It ships as a separate bundle, so we have to include the bundle in `index.html`.
+  
++makeExample('toh-6/ts/index.html', 'http', 'index.html (http)')(format=".")  
+  
+:marked
+  The only other thing we have to do is import `HTTP_PROVIDERS` from `angular2/http` and register it in `AppComponent`'s provider array.   
+  
++makeExample('toh-6/ts/app/app.component.ts', 'http', 'app/app.component (http)')(format=".")  
+
+:marked
+  ### Simulating the api
+  Our hero api is not deployed to production yet, but for the purposes of this chapter we have decided to integrate with an in-memory web api simulator.
+  
+  If you look at `AppComponent` you will notice that we have also imported `InMemoryBackendService`, `HeroData` and `SEED_DATA`. These imports are only there to configure the api simulator and will not be needed when integrating with a real api.
+  
+  We have included some more information about configuring the simulator <a href='/docs/ts/latest/guide/server-communication.html#!#in-mem-web-api'>here</a>.
+  
+  Since we are integrating with the simulator we no longer need `mock-heroes.ts`, so it's safe to delete it.
+
+:marked
+  ### Review the App Structure
+  Let’s verify that we have the following structure after all of our good refactoring in this chapter:
+  
+.filetree
+  .file angular2-tour-of-heroes
+  .children
+  .file app
+  .children
+    .file app.component.ts
+    .file app.component.css
+    .file dashboard.component.css
+    .file dashboard.component.html
+    .file dashboard.component.ts
+    .file hero.ts
+    .file hero-detail.component.css
+    .file hero-detail.component.html
+    .file hero-detail.component.ts
+    .file hero.service.ts
+    .file heroes.component.css
+    .file heroes.component.html
+    .file heroes.component.ts
+    .file main.ts
+    .file hero-data.ts
+  .file node_modules ...
+  .file typings ...      
+  .file index.html
+  .file package.json
+  .file styles.css
+  .file sample.css
+  .file tsconfig.json
+  .file typings.json
+    
+:marked
+
+.l-main-section
+:marked
+  ## Recap
+
```

Comment:
The story is good! Need a little more meat to wrap it up. We should be excited that we added http! 

Also, it may be nice to show the code all in a tabbed view at the end. I would love to do that in hindsight for all chapters.